crates/api/src/fetch.rs

```diff
@@ -0,0 +1,104 @@
+//! Kitsune2 fetch types.
+
+use std::sync::Arc;
+
+use prost::Message;
+
+use crate::{
+    builder, config, peer_store::DynPeerStore, transport::DynTransport,
+    AgentId, BoxFut, K2Error, K2Result, OpId, SpaceId,
+};
+
+include!("../proto/gen/kitsune2.fetch.rs");
+
+impl From<Vec<OpId>> for OpIds {
+    fn from(value: Vec<OpId>) -> Self {
+        Self {
+            data: value.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+impl From<OpIds> for Vec<OpId> {
+    fn from(value: OpIds) -> Self {
+        value.data.into_iter().map(Into::into).collect()
+    }
+}
+
+/// Serialize list of op ids for sending over the wire.
+pub fn serialize_op_ids(value: Vec<OpId>) -> bytes::Bytes {
+    let bytes = OpIds::from(value).encode_to_vec();
+    bytes::Bytes::copy_from_slice(&bytes)
+}
+
+/// Deserialize list of op ids.
+pub fn deserialize_op_ids(value: bytes::Bytes) -> K2Result<Vec<OpId>> {
+    let op_ids = OpIds::decode(value).map_err(K2Error::other)?;
+    let vec = Vec::from(op_ids);
+    Ok(vec)
+}
```

Comment:
Added functions to convert back and forth from serialized to deserialized op ids.